### ISSUE TITLE
apply fix to TemplateAction to im 4.0?

### ISSUE THREAD

**Issue by sergiocontrino** [intermine/intermine#2052]
in release 4.0 a bug has been introduced and commit c1bd944cdc8b24a8306df4f1b38f6ea4492e592e
in dev fixes it.
it has been fixed for flymine in production, and the 'one of' constraint seems to be working fine there after the fix.

**Comment by julie-sullivan** [comment]
From @rachellyne 

```

Take a look at this template, the one-of constraints do not work,

http://www.flymine.org/flymine/template.do?name=modENCODE_RNA_seq_Tissues
```

**Comment by julie-sullivan** [comment]
@sergiocontrino can you describe the bug that you fixed? What was the bad behaviour?

**Comment by julie-sullivan** [comment]
See #2044

**Comment by sergiocontrino** [comment]
the fix is in commit c1bd944, where the condition 
`String extraValue = extraValues == null ? null : extraValues[0];`

has been substituted by the verbose

`String extraValue;
if (extraValues == null) {
    extraValue = "";
    } else {
   extraValue = extraValues[0];                         
}`

extravalues being null was giving en error.

**Comment by julie-sullivan** [comment]
This is working in beta flymine.
